batch/batch/client.py

```diff
@@ -1,243 +1,131 @@
-                    volumes,
-                    security_context,
-                    service_account_name,
-                    attributes,
-                    batch_id,
-                    callback,
-                    parent_ids,
-                    input_files,
-                    output_files,
-                    always_run):
-        if env:
-            env = [{'name': k, 'value': v} for (k, v) in env.items()]
-        else:
-            env = []
-        env.extend([{
-            'name': 'POD_IP',
-            'valueFrom': {
-                'fieldRef': {'fieldPath': 'status.podIP'}
-            }
-        }, {
-            'name': 'POD_NAME',
-            'valueFrom': {
-                'fieldRef': {'fieldPath': 'metadata.name'}
-            }
-        }])
-
-        container = {
-            'image': image,
-            'name': 'main'
-        }
-        if command:
-            container['command'] = command
-        if args:
-            container['args'] = args
-        if env:
-            container['env'] = env
-        if ports:
-            container['ports'] = [{
-                'containerPort': p,
-                'protocol': 'TCP'
-            } for p in ports]
-        if resources:
-            container['resources'] = resources
-        if volumes:
-            container['volumeMounts'] = [v['volume_mount'] for v in volumes]
-        spec = {
-            'containers': [container],
-            'restartPolicy': 'Never'
-        }
-        if volumes:
-            spec['volumes'] = [v['volume'] for v in volumes]
-        if tolerations:
-            spec['tolerations'] = tolerations
-        if security_context:
-            spec['securityContext'] = security_context
-        if service_account_name:
-            spec['serviceAccountName'] = service_account_name
-
-        j = self.api.create_job(self.url, spec, attributes, batch_id, callback,
-                                parent_ids, input_files, output_files, always_run)
-        return Job(self,
-                   j['id'],
-                   attributes=j.get('attributes'),
-                   parent_ids=j.get('parent_ids', []))
-
-    def _get_job(self, id):
-        return self.api.get_job(self.url, id)
-
-    def _get_job_log(self, id):
-        return self.api.get_job_log(self.url, id)
-
-    def _get_batch(self, batch_id):
-        return self.api.get_batch(self.url, batch_id)
-
-    def _cancel_batch(self, batch_id):
-        self.api.cancel_batch(self.url, batch_id)
-
-    def _delete_batch(self, batch_id):
-        self.api.delete_batch(self.url, batch_id)
-
-    def _close_batch(self, batch_id):
-        return self.api.close_batch(self.url, batch_id)
+    def __init__(self, session, url=None, token_file=None, token=None, headers=None):
+        self._async_client = aioclient.BatchClient(session, url=url, token_file=token_file,
+                                                   token=token, headers=headers)
+
+    @property
+    def url(self):
```

Comment:
I don't think you need this? The url appears to be now only called from within aioclient restful methods.

Reply:
I thought it might be useful, but easy enough to add back in the future.